refactor(test): cleanup of L2 test structure and documentation

aliersh opened the pull request:
This PR refactors a portion of the test suite within the `/L2` folder as part of a broader effort to standardize structure, documentation, and formatting across the codebase.

### Changes applied to each file:
- Consolidated shared test initialization into dedicated `*_TestInit` contracts
- Organized test functions into logically separated contracts inheriting from `TestInit`
- Added `@title` and `@notice` tags to all test contracts
- Added function-level `@notice` comments describing expected behavior under test
- Replaced `@dev` tags with `@notice` where appropriate
- Enforced a 100-character limit for inline comments

### Progress

Refactored 25 of 25 test files (**100.0% complete**)

Note: `GasPriceOracle.t.sol` will be refactored in a later PR.

- [x] `BaseFeeVault.t.sol`
- [x] `CrossL2Inbox.t.sol`
- [x] `CrossDomainOwnable.t.sol`
- [x] `CrossDomainOwnable2.t.sol`
- [x] `CrossDomainOwnable3.t.sol`
- [x] `ETHLiquidity.t.sol`
- `GasPriceOracle.t.sol` ⏸ Deferred
- [x] `L1Block.t.sol`
- [x] `L1FeeVault.t.sol`
- [x] `L2CrossDomainMessenger.t.sol`
- [x] `L2ERC721Bridge.t.sol`
- [x] `L2StandardBridgeInterop.t.sol`
- [x] `L2StandardBridge.t.sol`
- [x] `L2ToL1MessagePasser.t.sol`
- [x] `L2ToL2CrossDomainMessenger.t.sol`
- [x] `OptimismMintableERC721.t.sol`
- [x] `OptimismMintableERC721Factory.t.sol`
- [x] `OptimismSuperchainERC20.t.sol`
- [x] `OptimismSuperchainERC20Beacon.t.sol`
- [x] `OptimismSuperchainERC20Factory.t.sol`
- [x] `OperatorFeeVault.t.sol`
- [x] `SequencerFeeVault.t.sol`
- [x] `SuperchainERC20.t.sol`
- [x] `SuperchainETHBridge.t.sol`
- [x] `SuperchainTokenBridge.t.sol`
- [x] `WETH.t.sol`

aliersh committed:
refactor(test): improve OperatorFeeVault test documentation

- rename test contract to OperatorFeeVault_Constructor_Test
- add @title and @notice tags to test contract
- convert @dev tag to @notice

aliersh committed:
refactor(test): improve L1FeeVault test documentation

- rename test contract to L1FeeVault_Constructor_Test
- add @title and @notice tags to test contract
- convert @dev tag to @notice

aliersh committed:
refactor(test): improve BaseFeeVault test documentation

- rename test contract to BaseFeeVault_Constructor_Test
- add @title and @notice tags to test contract
- convert @dev tag to @notice

aliersh committed:
refactor(test): improve OptimismSuperchainERC20Beacon test structure and documentation

- rename OptimismSuperchainERC20BeaconTest to OptimismSuperchainERC20Beacon_TestInit
- move implementation test into dedicated OptimismSuperchainERC20Beacon_Implementation_Test contract
- improve test contract documentation with more specific @title and @notice tags

aliersh committed:
refactor(test): improve WETH test structure and documentation

- organize individual test functions into separate contracts
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate

aliersh committed:
refactor(test): improve OptimismMintableERC721Factory test structure and documentation

- consolidate test initialization into a single OptimismMintableERC721Factory_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- add function-level @notice comments describing expected behavior under test
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve OptimismSuperchainERC20Factory test structure and documentation

- consolidate test initialization into a single OptimismSuperchainERC20Factory_TestInit contract
- organize individual test functions into separate contract inheriting from TestInit
- add @title and @notice tags to all test contracts

aliersh committed:
docs(test): improve test documentation clarity

- ensure comment lines stay within 100 character limit in OptimismSuperchainERC20Beacon test
- Update WETH test functions descriptions to explicitly mention default values

aliersh committed:
refactor(test): improve L2ToL1MessagePasser test structure and documentation

- organize individual test functions into separate contracts
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate

aliersh committed:
refactor(test): improve ExecutingMessageEmitted test structure and documentation

- consolidate test initialization into a single ExecutingMessageEmitted_TestInit contract
- rename ExecutingMessageEmittedTest to ExecutingMessageEmitted_Test
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve Predeploys test structure and documentation

- consolidate test initialization into a single Predeploys_TestInit contract
- rename PredeploysTest and PredeploysInteropTest to Predeploys_Test and Predeploys_Interop_Test
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve OptimismMintableERC721 test structure and documentation

- consolidate test initialization into a single OptimismMintableERC721_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- group tests not directly tied to a base contract function into OptimismMintableERC721_Unclassified_Test
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- add function-level @notice comments describing expected behavior under test
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve SuperchainERC20 test structure and documentation

- consolidate test initialization into a single SuperchainERC20_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve SuperchainETHBridge test structure and documentation

- consolidate test initialization into a single SuperchainETHBridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve ETHLiquidity test structure and documentation

- consolidate test initialization into a single ETHLiquidity_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
Merge branch 'ari/l2-tests-cleanup-part2' into ari/l2-tests-cleanups-part1

aliersh commented:
🔁 Continues work from the closed PR #15862, after renaming the branch.

aliersh committed:
refactor(test): improve SuperchainTokenBridge test structure and documentation

- consolidate test initialization into a single SuperchainTokenBridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve L1Block test structure and documentation

- consolidate test initialization into a single L1Block_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- add function-level @notice comments describing expected behavior under test
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve CrossL2Inbox test structure and documentation

- consolidate test initialization into a single CrossL2Inbox_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- add @notice tag to test function comments
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve L2StandardBridgeInterop test structure and documentation

- consolidate test initialization into a single L2StandardBridgeInterop_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve L2ERC721Bridge test structure and documentation

- consolidate test initialization into a single L2ERC721Bridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- group tests not directly tied to a base contract function into L2ERC721Bridge_Unclassified_Test
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve OptimismSuperchainERC20 test structure and documentation

- consolidate test initialization into a single OptimismSuperchainERC20_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- group tests not directly tied to a base contract function into OptimismSuperchainERC20_Unclassified_Test
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve L2CrossDomainMessenger test structure and documentation

- consolidate test initialization into a single L2CrossDomainMessenger_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- group tests not directly tied to a base contract function into L2CrossDomainMessenger_Unclassified_Test
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve L2ToL2CrossDomainMessenger test structure and documentation

- consolidate test initialization into a single L2ToL2CrossDomainMessenger_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): standardize test initialization contract documentation

- Update test initialization contract naming format to use _TestInit suffix
- Add backticks around contract names in documentation comments
- Fix duplicated notice tag in OptimismMintableERC721.t.sol
- Fix contract naming convention in ExecutingMessageEmitted.t.sol

aliersh commented on packages/contracts-bedrock/test/L2/OptimismMintableERC721.t.sol:
```diff
@@ -42,36 +48,30 @@ contract OptimismMintableERC721_Test is CommonTest {
+/// @notice Tests the `safeMint` function of the `OptimismMintableERC721` contract.
+contract OptimismMintableERC721_SafeMint_Test is OptimismMintableERC721_TestInit {
+    /// @notice Tests that the `safeMint` function successfully mints a token when called by the
+    /// @notice bridge.
```
Resolved in [3dd996a](https://github.com/ethereum-optimism/optimism/pull/15901/commits/3dd996a7d4e9068a1a1f38d2cd61616b2cb6802d)

aliersh commented on packages/contracts-bedrock/test/L2/ExecutingMessageEmitted.t.sol:
```diff
@@ -42,8 +43,14 @@ contract ExecutingMessageEmittedTest is CommonTest {
+}
 
-    /// @notice Tests that when `SuperchainERC20#crosschainMint` is called, the `ExecutingMessage` event is emitted.
+/// @title ExecutingMessageEmittedTest
```
Resolved in [3dd996a](https://github.com/ethereum-optimism/optimism/pull/15901/commits/3dd996a7d4e9068a1a1f38d2cd61616b2cb6802d)

aliersh committed:
revert(test): remove refactor of files not belonging to L2 cleanup scope

- remove Predeploys.t.sol refactor
- remove ExecutingMessageEmmited.t.sol refactor

aliersh committed:
refactor(test): improve CrossDomainOwnable test structure and documentation

- consolidate test initialization into a single CrossDomainOwnable_TestInit contract
- organize tests into logical groups based on functionality (AccessControl and PortalIntegration)
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve CrossDomainOwnable2 test structure and documentation

- consolidate test initialization into a single CrossDomainOwnable2_TestInit contract
- organize tests into logical groups (RevertConditions and SuccessConditions)
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve CrossDomainOwnable3 test structure and documentation

- consolidate test initialization into a single CrossDomainOwnable3_TestInit contract
- organize tests into logical groups (BasicFunctionality, TransferOwnership, and CrossDomainAccess)
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve L2StandardBridge test structure and documentation

- consolidate test initialization into a single L2StandardBridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- group tests not directly tied to a base contract function into L2StandardBridge_Unclassified_Test
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve SequencerFeeVault test structure and documentation

- consolidate test initialization into a single SequencerFeeVault_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

smartcontracts commented:
/ci authorize 61973cb490f2eef081954b3acff46630dd76b222

smartcontracts commented on packages/contracts-bedrock/test/L2/L2ToL2CrossDomainMessenger.t.sol:
```diff
@@ -368,14 +503,16 @@ contract L2ToL2CrossDomainMessengerTest is Test {
-    /// @dev Tests the `relayMessage` function returns the expected return data from the call to the target contract.
+    /// @notice Tests the `relayMessage` function returns the expected return data from the call to
+    ///         the target contract.
     function testFuzz_relayMessage_returnData_succeeds(
```
This test will need to be fixed, can explain how on next call

aliersh replied:
fixed in [0a1a07b](https://github.com/ethereum-optimism/optimism/pull/15901/commits/0a1a07b0332bbe571b6aa93625e2ebe9d10186a9)

aliersh committed:
fix(test): adds assumeNotForgeAddress to prevent Forge addresses as targets in L2ToL2CrossDomainMessenger tests

aliersh committed:
refactor(test): improve AssetReceiver test structure and documentation

- rena test Initializer contract to AssetReceiver_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- refactor test function names to follow naming convention
- add @title and @notice tags to test contracts
- ensure comment stay within 100 character limit

aliersh committed:
Revert "refactor(test): improve AssetReceiver test structure and documentation"

This reverts commit dd4a5a3473c8a7c006034521534728c23e9135f9.

smartcontracts commented:
/ci authorize 80f630e1885b2b558b1daa33d8f35bf96f646944
